property.go

```diff
@@ -125,10 +122,13 @@ func getPropertyName(expr ast.Expr, parser *Parser) (propertyName, error) {
 }
 
 func getArrayPropertyName(astTypeArray *ast.ArrayType, parser *Parser) propertyName {
-	if astTypeArrayExpr, ok := astTypeArray.Elt.(*ast.SelectorExpr); ok {
+	if _, ok := astTypeArray.Elt.(*ast.StructType); ok {
+		return propertyName{SchemaType: "array", ArrayType: "object"}
+	} else if _, ok := astTypeArray.Elt.(*ast.MapType); ok {
+		return propertyName{SchemaType: "array", ArrayType: "object"}
+	} else if astTypeArrayExpr, ok := astTypeArray.Elt.(*ast.SelectorExpr); ok {
 		return parseFieldSelectorExpr(astTypeArrayExpr, parser, newArrayProperty)
-	}
-	if astTypeArrayExpr, ok := astTypeArray.Elt.(*ast.StarExpr); ok {
+	} else if astTypeArrayExpr, ok := astTypeArray.Elt.(*ast.StarExpr); ok {
```

Comment:
This part of the code seems like a little messy.  It should be refactored in a good way in the future.